.github/workflows/ci.yml

```diff
@@ -1,76 +1,79 @@
 name: CI
 
-# These trigger patterns courtesy of https://github.com/broccolijs/broccoli/pull/436
 on:
-  pull_request:
   push:
-    # filtering branches here prevents duplicate builds from pull_request and push
     branches:
       - main
-      - "v*"
-    # always run CI for tags
-    tags:
-      - "*"
+      - master
+  pull_request: {}
 
-env:
-  CI: true
+concurrency:
+  group: ci-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
 
 jobs:
-  test-locked-deps:
-    name: Locked Deps
+  test:
+    name: "Tests"
     runs-on: ubuntu-latest
+    timeout-minutes: 10
+
     steps:
-      - name: Checkout Code
-        uses: actions/checkout@v2.3.4
+      - uses: actions/checkout@v3
       - name: Install Node
-        uses: actions/setup-node@v2.1.5
+        uses: actions/setup-node@v3
         with:
-          node-version: 14
+          node-version: 12.x
+          cache: yarn
       - name: Install Dependencies
         run: yarn install --frozen-lockfile
-      - name: Browser Tests
-        run: yarn test
+      - name: Lint
+        run: yarn lint
+      - name: Run Tests
+        run: yarn test:ember
 
-  test-old-dependencies:
-    name: Oldest Supported Env
+  floating:
+    name: "Floating Dependencies"
     runs-on: ubuntu-latest
+    timeout-minutes: 10
+
     steps:
-      - name: Checkout Code
-        uses: actions/checkout@v2.3.4
-      - name: Install Node
-        uses: actions/setup-node@v2.1.5
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v3
         with:
-          node-version: 10
+          node-version: 12.x
+          cache: yarn
       - name: Install Dependencies
-        run: yarn install --frozen-lockfile
-      - name: Browser Tests
-        run: yarn test
+        run: yarn install --no-lockfile
+      - name: Run Tests
+        run: yarn test:ember
 
-  test-try:
-    name: Ember Try
+  try-scenarios:
+    name: ${{ matrix.try-scenario }}
     runs-on: ubuntu-latest
-    needs: [test-locked-deps]
-    continue-on-error: ${{ matrix.experimental }}
+    needs: "test"
+    timeout-minutes: 10
+
     strategy:
       fail-fast: false
       matrix:
-        scenario:
-          - ember-lts-2.8
-          - ember-lts-2.12
+        try-scenario:
+          - ember-lts-3.24
+          - ember-lts-3.28
+          - ember-4.0
           - ember-release
           - ember-beta
-        experimental: [false]
-        include:
-          - scenario: ember-canary
-            experimental: true
+          - ember-canary
+          # - embroider-safe
+          # - embroider-optimized
```

Comment:
you can enable these with the changes i mention in the ember try file

Reply:
embroider-optimized fails: https://github.com/ember-intl/cp-validations/actions/runs/3096110587/jobs/5011285918

Reply:
Even when given the configuration?

```
embroiderOptimized({
  npm: {
    devDependencies: {
      'ember-auto-import': '^2',
      webpack: '^5',
    },
  },
}),
```
I wonder if it is from something upstream..

Reply:
```js
const specifier = '@ember-intl/cp-validations@validator:messages';
const instance = this.owner.lookup(specifier);
instance.getMessageFor('foobarbaz');
```

seems like `instance` is undefined in`embroider-optimized` scenario.

```js
function embroiderOptimized(extension) {
    return extendScenario({
        name: 'embroider-optimized',
        npm: {
            devDependencies: {
                '@embroider/core': currentEmbroiderVersion,
                '@embroider/webpack': currentEmbroiderVersion,
                '@embroider/compat': currentEmbroiderVersion,
                // Webpack is a peer dependency of `@embroider/webpack`
                webpack: '^5.0.0',
            },
        },
        env: {
            EMBROIDER_TEST_SETUP_OPTIONS: 'optimized',
        },
    }, extension);
}
```

I think `ember-auto-import` `v2` is present via `@embroider/util` which has `"ember-auto-import": "^2.2.0",` as a dependency. Also the `embroider-safe` scenario works which has the same `devDependencies` as `embroiderOptimized` from above.

```
$ npm ls @embroider/util
@ember-intl/cp-validations@5.0.0 /Users/manuel/Code/opensource/cp-validations
└─┬ @ember/test-helpers@2.8.1
  └── @embroider/util@1.8.3
```

I'll create a new issue with the information. If you want to try to fix it a PR is very welcome.

Reply:
Tracked in https://github.com/ember-intl/cp-validations/issues/92